ehsm_kms_service/ehsm_kms_server.js

```diff
@@ -4,10 +4,16 @@ const app = express();
 const logger = require('./logs/logger');
 const ehsm_napi = require('./ehsm_napi');
 const { ehsm_kms_params, ehsm_keyspec_t, ehsm_keyorigin_t} = require('./ehsm_kms_params.js')
-
+const userInfo = require('./user_info')
 app.use(express.json());
 
-const port = 9000;
+const PORT = process.argv.slice(2)[0] || 9000;
+
+const NONCE_CACHE_TIME = 15 * 60 * 1000;
```

Comment:
As the replay attack I mentioned, the nonce should be kept in the database for 2*MAX_TIME_STAMP_DIFF.

Reply:
the MAX_TIME_STAMP_DIFF will be changed to 5 minutes,It is a typo.

Reply:
explicitly show the relationship between MAX_TIME_STAMP_DIFF and NONCE_CACHE_TIME:
NONCE_CACHE_TIME = 2 * MAX_TIME_STAMP_DIFF

Reply:
seems the logic is too complicated!

we can use the timestamp only. it can also prevent the replay attack.